src/components/structures/MatrixChat.js

```diff
@@ -628,6 +634,17 @@ module.exports = React.createClass({
         this.showScreen("settings");
     },
 
+    updateFavicon: function() {
+        var notifCount = 0;
+
+        var rooms = MatrixClientPeg.get().getRooms();
+        for (var i = 0; i < rooms.length; ++i) {
+            notifCount += rooms[i].unread_notification_count;
```

Comment:
May want an `undefined` guard here to avoid risk of `NaN`.

```javascript
if (rooms[i].unread_notification_count) {
  notifCount += rooms[i].unread_notification_count;
}
```